staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -505,8 +505,7 @@ func diffListsOfMaps(original, modified []interface{}, t reflect.Type, mergeKey
 func getMapAndMergeKeyValueByIndex(index int, mergeKey string, listOfMaps []interface{}) (map[string]interface{}, interface{}, error) {
 	m, ok := listOfMaps[index].(map[string]interface{})
 	if !ok {
-		t := reflect.TypeOf(listOfMaps[index])
-		return nil, nil, mergepatch.ErrBadArgType("map[string]interface{}", t.Kind().String())
+		return nil, nil, mergepatch.ErrBadArgType(m, listOfMaps[index])
```

Comment:
I think so.
If `listOfMaps[index]` is `map[int]interface{}`, but not `map[string]interface{}`, this is bad type, even though their kinds are both `Map`.